### ISSUE TITLE
ZSQLmethods broken in master

### ISSUE THREAD

**Issue by garrysadd** [zopefoundation/Zope#316]
When adding a ZSQLmethod I get this:

Resource not found

Sorry, the requested resource does not exist.
Check the URL and try again.

Resource: manage_manage_addZSQLMethod POST

Anyone know how to get round this?

**Comment by tseaver** [comment]
Duplicate of https://github.com/zopefoundation/Products.ZSQLMethods/issues/9.

**Comment by viktordick** [comment]
The problem is not in ZSQLMethods after all. It stems from the handling of modal dialogs for adding objects, which trys to replace the form action. It breaks for ZSQLMethods - the URL is changed from `manage_addZSQLMethod` to `http://localhost/manage_manage_addZSQLMethod`.

A related problem occurs if adding a python script, closing the modal dialog and reopening it - the URL is rewritten twice, which makes it invalid.

In summary, the handling of modal dialogs and its URL rewriting needs to be fixed.
